kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerInstanceManager.java

```diff
@@ -254,30 +282,63 @@ public Void doOperation(KieServicesClient client,
                                             });
     }
 
-    public List<Container> upgradeContainer(ServerTemplate serverTemplate,
+    public List<Container> upgradeContainer(final ServerTemplate serverTemplate,
                                             final ContainerSpec containerSpec) {
 
         return callRemoteKieServerOperation(serverTemplate,
                                             containerSpec,
-                                            new RemoteKieServerOperation<Void>() {
-                                                @Override
-                                                public Void doOperation(KieServicesClient client,
-                                                                        Container container) {
+                                            makeUpgradeContainerOperation(containerSpec));
+    }
 
-                                                    ServiceResponse<ReleaseId> response = client.updateReleaseId(containerSpec.getId(),
-                                                                                                                 containerSpec.getReleasedId());
-                                                    if (!response.getType().equals(ServiceResponse.ResponseType.SUCCESS)) {
-                                                        logger.debug("Container {} failed to upgrade on server instance {} due to {}",
-                                                                     containerSpec.getId(),
-                                                                     container.getUrl(),
-                                                                     response.getMsg());
-                                                    }
-                                                    collectContainerInfo(containerSpec,
-                                                                         client,
-                                                                         container);
-                                                    return null;
-                                                }
-                                            });
+    public List<Container> upgradeAndStartContainer(final ServerTemplate serverTemplate,
```

Comment:
Does this method have sense? When container is started (created on Kie server side) then all configurations are applied. So AFAIK this method has same functionality as startContainer(...).

Reply:
@sutaakar Indeed. The only difference is in the `messages` section, where instead of `Release id successfully updated for container {containerName}`, the user will see `Container {containerId} successfully created with module {releaseId}`. I'm not sure if this is a problem, but if it's not, I think that we can replace the `upgradeAndStartContainer ` calls by `startContainer`.